### ISSUE TITLE
$field->slug() and Str::slug() don't use custom slug rules

### ISSUE THREAD

**Issue by neildaniels** [getkirby/kirby#2535]
**Describe the bug**  
`$field->slug()` (and its underlying `Str::slug()` usage) doesn't take into account custom slug rules that may have been set up for a language.

**To Reproduce**  
1. Set up [language-specific slugs](https://getkirby.com/docs/guide/languages/slug-rules)
2. Use `$field->slug()` on something

**Expected behavior**  
I would definitely expect `$field->slug()` to be aware of the current language's rules. 

For `Str::slug()`, I would expect you to be to specify a language (or pass in a custom array of slugs).

**Kirby Version**  
`3.3.5`

**Additional context**  
Current workaround is a pretty simple `Str::replace()`.

```php
$slug = (string)$someField;
if ($language->slugs()) {
    $slug = Str::replace($slug, array_keys($language->slugs()), array_values($language->slugs()));
}
$slug = Str::slug($slug);
```

**Comment by afbora** [comment]
May be related #2230

**Comment by afbora** [comment]
Fixed on 3.5.0 with #2880 PR
